plugins/uartdmx/UartWidget.cpp

```diff
@@ -122,6 +124,10 @@ bool UartWidget::Write(const ola::DmxBuffer& data) {
   buffer[0] = DMX512_START_CODE;
 
   data.Get(buffer + 1, &length);
+  if (length < m_padding) {
+      memset((buffer + 1 + length), 0x00, (m_padding - length) );
+      length = m_padding;
+    }
```

Comment:
Given as you've said this is probably required in a few other plugins, I wonder if it makes sense to add some extra options to DmxBuffer? e.g. GetWithPadding(uint8_t *data, unsigned int *length, unsigned int min_length). Or alternatively a Pad/Grow function which modifies the buffer in place (to save duplicating lots of functions), although the latter would need to be called on the lower level buffer in UartThread as this one is const so maybe GetWithPadding is simpler.